pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -775,10 +787,28 @@ func getLocationFromPod(clientset kubernetes.Interface, d *apps.Deployment) (str
 			hostName = pvcName
 		}
 	}
-	return GetLocationWithNode(clientset, nodeName, hostName)
+	// compatibility with existing clusters
```

Comment:
Instead of finding the root on the osd pod, in this scenario I'd suggest the crushRoot be passed into this method since it's a known property on the CR. Only properties that the operator can't detect (e.g. hostName) need to be looked up on the existing pod.